src/JsRoot/JsRootGraph.tsx

```diff
@@ -1,38 +1,50 @@
-import React, { useEffect, useRef, useState } from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
+import CircularProgress from '@mui/material/CircularProgress';
 import { useJSROOT } from './JsRootService';
+import { useVisible } from 'react-hooks-visible'
+
 interface JsRootGraphProps {
-    data?: Object
+    data?: any
 }
 
 function JsRootGraph(props: JsRootGraphProps) {
     const { JSROOT } = useJSROOT();
+    const [containerEl, visible] = useVisible<HTMLDivElement>();
 
-    const containerEl = useRef<HTMLDivElement>(null);
     const [obj, setObj] = useState(undefined);
+    const [drawn, setDrawn] = useState(false);
 
 
     useEffect(() => {
         // create example graph
-        const nbinsx = 20;
-        const y = Array.from({ length: nbinsx }, () => Math.floor(Math.random() * 40));
+        const nbinsx = props.data?.value ?? 20;
+        const y = Array.from({ length: nbinsx }, () => Math.floor(Math.random() * 20));
         const x = Array.from({ length: nbinsx }, (v, k) => k);
 
         const h1 = JSROOT.createTGraph(nbinsx, x, y);
```

Comment:
Rename `h1` to `gr` to follow ROOT conventions:
https://root.cern.ch/doc/master/classTGraph.html

`h1` suggests we have histogram here